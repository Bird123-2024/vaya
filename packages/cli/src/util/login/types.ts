export interface LoginData {
  token: string;
  securityCode: string;
}

export interface VerifyData {
  email: string;
  token: string;
}

export interface SAMLError {
  saml?: true;
  teamId: string | null;
  scope: string;
  enforced?: boolean;
}
