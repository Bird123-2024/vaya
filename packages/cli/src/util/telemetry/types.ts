import type { Prettify, ToTitleCase } from '../types';
import type {
  Command,
  CommandArgument,
  CommandOption,
  PrimitiveConstructor,
} from '../../commands/help';

// ToTelemetryOptionName<{ name: 'foo', type: BooleanConstructor }> → 'trackCliFlagFoo'
// ToTelemetryOptionName<{ name: 'foo', type: StringConstructor }> → 'trackCliOptionFoo'
// ToTelemetryOptionName<{ name: 'foo', type: [StringConstructor] }> → 'trackCliOptionFoo'
type ToTelemetryOptionName<T extends CommandOption> =
  T['type'] extends BooleanConstructor
    ? `trackCliFlag${ToTitleCase<T['name']>}`
    : T['type'] extends StringConstructor
      ? `trackCliOption${ToTitleCase<T['name']>}`
      : T['type'] extends ReadonlyArray<StringConstructor>
        ? `trackCliOption${ToTitleCase<T['name']>}`
        : never;

// ToArgType<BooleanConstructor> → boolean
// ToArgType<StringConstructor> → string
// ToArgType<[StringConstructor]> → [string]
type ToArgType<T extends CommandOption['type']> = T extends readonly any[]
  ? [ToArgType<T[0]>]
  : T extends PrimitiveConstructor
    ? ReturnType<T>
    : never;

// TelemetryOptionMethods<{ name: 'foo', type: BooleanConstructor }> → { trackCliOptionFoo: (v: boolean | undefined) => void; }
type TelemetryOptionMethods<A extends readonly CommandOption[]> = {
  [K in A[number] as ToTelemetryOptionName<K>]: (
    v: ToArgType<K['type']> | undefined
  ) => void;
};

// TelemetryArgumentMethods<{ name: 'foo' }> → { trackCliArgumentFoo: (v: string | undefined) => void; }
type TelemetryArgumentMethods<A extends readonly CommandArgument[]> = {
  [K in A[number] as `trackCliArgument${ToTitleCase<K['name']>}`]: (
    v: string | undefined
  ) => void;
};

export type TelemetryMethods<T extends Command> = Prettify<
  TelemetryArgumentMethods<T['arguments']> &
    TelemetryOptionMethods<T['options']>
>;
