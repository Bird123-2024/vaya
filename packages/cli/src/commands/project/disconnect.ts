import chalk from 'chalk';
import { Org, Project, Team } from '../../types';
import Client from '../../util/client';
import confirm from '../../util/input/confirm';
import { getCommandName } from '../../util/pkg-name';
import { disconnectGitProvider } from '../../util/projects/connect-git-provider';

export default async function disconnect(
  client: Client,
  args: string[],
  project: Project | undefined,
  org: Org | undefined,
  team: Team | null
) {
  const { output } = client;

  if (args.length !== 0) {
    output.error(
      `Invalid number of arguments. Usage: ${chalk.cyan(
        `${getCommandName('project disconnect')}`
      )}`
    );
    return 2;
  }
  if (!project || !org) {
    output.error('An unexpected error occurred.');
    return 1;
  }

  if (project.link) {
    const { org, repo } = project.link;
    output.print(
      `Your Vercel project will no longer create deployments when you push to this repository.\n`
    );
    const confirmDisconnect = await confirm(
      client,
      `Are you sure you want to disconnect ${chalk.cyan(
        `${org}/${repo}`
      )} from your project?`,
      false
    );

    if (confirmDisconnect) {
      await disconnectGitProvider(client, team, project.id);
      output.log(`Disconnected ${chalk.cyan(`${org}/${repo}`)}.`);
    } else {
      output.log('Aborted.');
    }
  } else {
    output.error(
      `No Git repository connected. Run ${getCommandName(
        'project connect'
      )} to connect one.`
    );
    return 1;
  }

  return 0;
}
