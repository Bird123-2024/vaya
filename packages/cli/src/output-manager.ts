import { Output } from './util/output';
import { OutputOptions } from './util/output/create-output';

let output = new Output(process.stderr, {
  debug: false,
  noColor: true,
});

export function outputInit(options: OutputOptions) {
  output.initialize(options);
}

/**
 * A managed singleton instance of the Output class.
 */
export default output;
