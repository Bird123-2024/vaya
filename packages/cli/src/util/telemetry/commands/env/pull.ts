import { TelemetryClient } from '../..';

export class EnvPullTelemetryClient extends TelemetryClient {
  trackCliArgumentFilename(filename?: string) {
    if (filename) {
      this.trackCliArgument({
        arg: 'filename',
        value: this.redactedValue,
      });
    }
  }

  trackCliOptionEnvironment(environment?: string) {
    if (environment) {
      this.trackCliOption({
        option: 'environment',
        value: this.redactedValue,
      });
    }
  }

  trackCliOptionGitBranch(gitBranch?: string) {
    if (gitBranch) {
      this.trackCliOption({
        option: 'git-branch',
        value: this.redactedValue,
      });
    }
  }

  trackCliFlagYes(yes?: boolean) {
    if (yes) {
      this.trackCliFlag('yes');
    }
  }
}
