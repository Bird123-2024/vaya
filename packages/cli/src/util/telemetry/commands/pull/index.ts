import { TelemetryClient } from '../..';
import type { TelemetryMethods } from '../../types';
import type { pullCommand } from '../../../../commands/pull/command';

export class PullTelemetryClient
  extends TelemetryClient
  implements TelemetryMethods<typeof pullCommand>
{
  trackCliArgumentProjectPath: (v: string | undefined) => void;

  trackCliOptionEnvironment(environment?: string) {
    if (environment) {
      const standardEnvironments = ['production', 'preview', 'development'];
      this.trackCliOption({
        option: 'environment',
        value: standardEnvironments.includes(environment)
          ? environment
          : this.redactedValue,
      });
    }
  }

  trackCliOptionGitBranch(gitBranch?: string) {
    if (gitBranch) {
      this.trackCliOption({
        option: 'git-branch',
        value: this.redactedValue,
      });
    }
  }

  trackCliFlagProd(isProduction?: boolean) {
    if (isProduction) {
      this.trackCliFlag('prod');
    }
  }

  trackCliFlagYes(yes?: boolean) {
    if (yes) {
      this.trackCliFlag('yes');
    }
  }
}
