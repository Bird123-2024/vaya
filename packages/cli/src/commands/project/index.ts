import chalk from 'chalk';
import Client from '../../util/client';
import { ensureLink } from '../../util/ensure-link';
import getArgs from '../../util/get-args';
import getInvalidSubcommand from '../../util/get-invalid-subcommand';
import getScope from '../../util/get-scope';
import handleError from '../../util/handle-error';
import logo from '../../util/output/logo';
import { getPkgName } from '../../util/pkg-name';
import validatePaths from '../../util/validate-paths';
import add from './add';
import connect from './connect';
import disconnect from './disconnect';
import list from './list';
import rm from './rm';

const help = () => {
  console.log(`
  ${chalk.bold(`${logo} ${getPkgName()} project`)} [options] <command>

  ${chalk.dim('Commands:')}

    ls                               Show all projects in the selected team/user
    connect                          Connect a Git provider repository to your project
    disconnect                       Disconnect a Git provider repository from your project
    add      [name]                  Add a new project
    rm       [name]                  Remove a project

  ${chalk.dim('Options:')}

    -h, --help                     Output usage information
    -t ${chalk.bold.underline('TOKEN')}, --token=${chalk.bold.underline(
    'TOKEN'
  )}        Login token
    -S, --scope                    Set a custom scope
    -N, --next                     Show next page of results

  ${chalk.dim('Examples:')}

  ${chalk.gray('–')} Add a new project

    ${chalk.cyan(`$ ${getPkgName()} project add my-project`)}

  ${chalk.gray('–')} Paginate projects, where ${chalk.dim(
    '`1584722256178`'
  )} is the time in milliseconds since the UNIX epoch.

    ${chalk.cyan(`$ ${getPkgName()} project ls --next 1584722256178`)}
`);
};

const COMMAND_CONFIG = {
  ls: ['ls', 'list'],
  add: ['add'],
  rm: ['rm', 'remove'],
  connect: ['connect'],
};

export default async function main(client: Client) {
  let argv: any;
  let subcommand: string | string[];

  try {
    argv = getArgs(client.argv.slice(2), {
      '--next': Number,
      '-N': '--next',
      '--confirm': Boolean,
    });
  } catch (error) {
    handleError(error);
    return 1;
  }

  if (argv['--help']) {
    help();
    return 2;
  }

  argv._ = argv._.slice(1);
  subcommand = argv._[0] || 'list';
  const args = argv._.slice(1);
  const confirm = Boolean(argv['--confirm']);
  const { output } = client;

  let paths = [process.cwd()];
  const pathValidation = await validatePaths(client, paths);
  if (!pathValidation.valid) {
    return pathValidation.exitCode;
  }
  const { path } = pathValidation;

  let org;
  let project;
  let contextName = '';
  let team = null;

  if (subcommand === 'connect' || subcommand === 'disconnect') {
    const linkedProject = await ensureLink('project', client, path, confirm);
    if (typeof linkedProject === 'number') {
      return linkedProject;
    }
    org = linkedProject.org;
    project = linkedProject.project;
  } else {
    try {
      ({ contextName, team } = await getScope(client));
    } catch (error) {
      if (error.code === 'NOT_AUTHORIZED' || error.code === 'TEAM_DELETED') {
        output.error(error.message);
        return 1;
      }
      throw error;
    }
  }

  switch (subcommand) {
    case 'ls':
    case 'list':
      return await list(client, argv, args, contextName);
    case 'add':
      return await add(client, args, contextName);
    case 'rm':
    case 'remove':
      return await rm(client, args);
    case 'connect':
      return await connect(client, argv, args, project, org, team);
    case 'disconnect':
      return await disconnect(client, args, project, org, team);
    default:
      output.error(getInvalidSubcommand(COMMAND_CONFIG));
      help();
      return 2;
  }
}
