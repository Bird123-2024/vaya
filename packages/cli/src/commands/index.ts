import { aliasCommand } from './alias/command';
import { bisectCommand } from './bisect/command';
import { buildCommand } from './build/command';
import { certsCommand } from './certs/command';
import { deployCommand } from './deploy/command';
import { devCommand } from './dev/command';
import { dnsCommand } from './dns/command';
import { domainsCommand } from './domains/command';
import { envCommand } from './env/command';
import { gitCommand } from './git/command';

import { initCommand } from './init/command';
import { inspectCommand } from './inspect/command';
import { installCommand } from './install/command';
import { integrationResourceCommand } from './integration-resource/command';
import { integrationCommand } from './integration/command';
import { linkCommand } from './link/command';
import { listCommand } from './list/command';
import { loginCommand } from './login/command';
import { logoutCommand } from './logout/command';
import { logsCommand } from './logs/command';
import { projectCommand } from './project/command';
import { promoteCommand } from './promote/command';
import { pullCommand } from './pull/command';
import { redeployCommand } from './redeploy/command';
import { removeCommand } from './remove/command';
import { rollbackCommand } from './rollback/command';
import { targetCommand } from './target/command';
import { teamsCommand } from './teams/command';
import { telemetryCommand } from './telemetry/command';
import { whoamiCommand } from './whoami/command';
import type { Command } from './help';

const commands = [
  aliasCommand,
  bisectCommand,
  buildCommand,
  certsCommand,
  deployCommand,
  devCommand,
  dnsCommand,
  domainsCommand,
  envCommand,
  gitCommand,
  initCommand,
  inspectCommand,
  installCommand,
  integrationCommand,
  integrationResourceCommand,
  linkCommand,
  listCommand,
  loginCommand,
  logoutCommand,
  logsCommand,
  projectCommand,
  promoteCommand,
  pullCommand,
  redeployCommand,
  removeCommand,
  rollbackCommand,
  targetCommand,
  teamsCommand,
  telemetryCommand,
  whoamiCommand,
  // added because we don't have a full help command
  { name: 'help', aliases: [] },
];

export function getAliases(command: Pick<Command, 'name' | 'aliases'>) {
  return [command.name].concat(command.aliases);
}

export function getCommandAliasMap() {
  const aliasMap = new Map();
  for (const command of commands) {
    const aliases = getAliases(command);
    for (const alias of aliases) {
      aliasMap.set(alias, command.name);
    }
  }
  return aliasMap;
}

export default new Map([
  ['alias', 'alias'],
  ['aliases', 'alias'],
  ['bisect', 'bisect'],
  ['build', 'build'],
  ['cert', 'certs'],
  ['certs', 'certs'],
  ['deploy', 'deploy'],
  ['dev', 'dev'],
  ['develop', 'dev'],
  ['dns', 'dns'],
  ['domain', 'domains'],
  ['domains', 'domains'],
  ['env', 'env'],
  ['git', 'git'],
  ['help', 'help'],
  ['i', 'install'],
  ['init', 'init'],
  ['inspect', 'inspect'],
  ['install', 'install'],
  ['integration', 'integration'],
  ['integration-resource', 'integration-resource'],
  ['ir', 'integration-resource'],
  ['link', 'link'],
  ['list', 'list'],
  ['ln', 'alias'],
  ['log', 'logs'],
  ['login', 'login'],
  ['logout', 'logout'],
  ['logs', 'logs'],
  ['ls', 'list'],
  ['project', 'project'],
  ['projects', 'project'],
  ['promote', 'promote'],
  ['pull', 'pull'],
  ['redeploy', 'redeploy'],
  ['remove', 'remove'],
  ['rm', 'remove'],
  ['rollback', 'rollback'],
  ['switch', 'teams'],
  ['target', 'target'],
  ['targets', 'target'],
  ['team', 'teams'],
  ['teams', 'teams'],
  ['telemetry', 'telemetry'],
  ['whoami', 'whoami'],
]);
