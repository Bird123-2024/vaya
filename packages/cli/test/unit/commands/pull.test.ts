import fs from 'fs-extra';
import path from 'path';
import pull from '../../../src/commands/pull';
import { setupFixture } from '../../helpers/setup-fixture';
import { client } from '../../mocks/client';
import { defaultProject, useProject } from '../../mocks/project';
import { useTeams } from '../../mocks/team';
import { useUser } from '../../mocks/user';
import { useDeploymentMissingProjectSettings } from '../../mocks/deployment';
import { Project } from '../../../src/types';

describe('pull', () => {
  it('should handle pulling', async () => {
    const cwd = setupFixture('vercel-pull-next');
    useUser();
    useTeams('team_dummy');
    useProject({
      ...defaultProject,
      id: 'vercel-pull-next',
      name: 'vercel-pull-next',
    });
    client.setArgv('pull', cwd);
    const exitCodePromise = pull(client);
    await expect(client.stderr).toOutput(
      'Downloading `development` Environment Variables for Project vercel-pull-next'
    );
    await expect(client.stderr).toOutput(
      `Created .vercel${path.sep}.env.development.local file`
    );
    await expect(client.stderr).toOutput(
      `Downloaded project settings to .vercel${path.sep}project.json`
    );
    await expect(exitCodePromise).resolves.toEqual(0);

    const rawDevEnv = await fs.readFile(
      path.join(cwd, '.vercel', '.env.development.local')
    );
    const devFileHasDevEnv = rawDevEnv.toString().includes('SPECIAL_FLAG');
    expect(devFileHasDevEnv).toBeTruthy();
  });

  it('should fail with message to pull without a link and without --env', async () => {
    client.stdin.isTTY = false;

    const cwd = setupFixture('vercel-pull-unlinked');
    useUser();
    useTeams('team_dummy');

    client.setArgv('pull', cwd);
    const exitCodePromise = pull(client);
    await expect(client.stderr).toOutput(
      'Command `vercel pull` requires confirmation. Use option "--yes" to confirm.'
    );
    await expect(exitCodePromise).resolves.toEqual(1);
  });

  it('should fail without message to pull without a link and with --env', async () => {
    const cwd = setupFixture('vercel-pull-next');
    useUser();
    useTeams('team_dummy');

    client.setArgv('pull', cwd, '--yes');
    const exitCodePromise = pull(client);
    await expect(client.stderr).not.toOutput(
      'Command `vercel pull` requires confirmation. Use option "--yes" to confirm.'
    );
    await expect(exitCodePromise).resolves.toEqual(1);
  });

  it('should link an unlinked project with a git config without connecing a git provider', async () => {
    const cwd = setupFixture('vercel-pull-unlinked-git');

    try {
      await fs.rename(path.join(cwd, 'git'), path.join(cwd, '.git'));

      let project: Project;
      let projectWasLinked = false;

      client.scenario.get(`/v8/projects/:projectNameOrId`, (_req, res) => {
        res.status(404).send();
      });
      client.scenario.post(`/:version/projects`, (req, res) => {
        const { name } = req.body;
        project = {
          ...defaultProject,
          name,
          id: name,
        };
        res.json(project);
      });
      client.scenario.post(`/v9/projects/:projectNameOrId/link`, (req, res) => {
        projectWasLinked = true;

        const { type, repo, org } = req.body;
        project.link = {
          type,
          repo,
          repoId: 1010,
          org,
          gitCredentialId: '',
          sourceless: true,
          createdAt: 1656109539791,
          updatedAt: 1656109539791,
        };
        res.json(project);
      });
      client.scenario.patch(
        `/:version/projects/:projectNameOrId`,
        (req, res) => {
          Object.assign(project, req.body);
          res.json(project);
        }
      );
      client.scenario.get(`/v8/projects/:id/env`, (_req, res) => {
        res.json({ envs: [] });
      });

      useDeploymentMissingProjectSettings();
      useUser();
      useTeams('team_dummy');

      client.setArgv('pull', cwd, '--yes');
      const exitCodePromise = pull(client);

      await expect(exitCodePromise).resolves.toEqual(0);
      expect(projectWasLinked).toBe(false);
    } finally {
      await fs.rename(path.join(cwd, '.git'), path.join(cwd, 'git'));
    }
  });

  it('should handle pulling with env vars (headless mode)', async () => {
    try {
      process.env.VERCEL_PROJECT_ID = 'vercel-pull-next';
      process.env.VERCEL_ORG_ID = 'team_dummy';

      const cwd = setupFixture('vercel-pull-next');

      // Remove the `.vercel` dir to ensure that the `pull`
      // command creates a new one based on env vars
      await fs.remove(path.join(cwd, '.vercel'));

      useUser();
      useTeams('team_dummy');
      useProject({
        ...defaultProject,
        id: 'vercel-pull-next',
        name: 'vercel-pull-next',
      });
      client.setArgv('pull', cwd);
      const exitCodePromise = pull(client);
      await expect(client.stderr).toOutput(
        'Downloading `development` Environment Variables for Project vercel-pull-next'
      );
      await expect(client.stderr).toOutput(
        `Created .vercel${path.sep}.env.development.local file`
      );
      await expect(client.stderr).toOutput(
        `Downloaded project settings to .vercel${path.sep}project.json`
      );
      await expect(exitCodePromise).resolves.toEqual(0);

      const config = await fs.readJSON(path.join(cwd, '.vercel/project.json'));
      expect(config).toMatchInlineSnapshot(`
        Object {
          "orgId": "team_dummy",
          "projectId": "vercel-pull-next",
          "settings": Object {
            "createdAt": 1555413045188,
          },
        }
      `);
    } finally {
      delete process.env.VERCEL_PROJECT_ID;
      delete process.env.VERCEL_ORG_ID;
    }
  });

  it('should handle --environment=preview flag', async () => {
    const cwd = setupFixture('vercel-pull-next');
    useUser();
    useTeams('team_dummy');
    useProject({
      ...defaultProject,
      id: 'vercel-pull-next',
      name: 'vercel-pull-next',
    });
    client.setArgv('pull', '--environment=preview', cwd);
    const exitCodePromise = pull(client);
    await expect(client.stderr).toOutput(
      'Downloading `preview` Environment Variables for Project vercel-pull-next'
    );
    await expect(client.stderr).toOutput(
      `Created .vercel${path.sep}.env.preview.local file`
    );
    await expect(client.stderr).toOutput(
      `Downloaded project settings to .vercel${path.sep}project.json`
    );
    await expect(exitCodePromise).resolves.toEqual(0);

    const rawPreviewEnv = await fs.readFile(
      path.join(cwd, '.vercel', '.env.preview.local')
    );
    const previewFileHasPreviewEnv = rawPreviewEnv
      .toString()
      .includes('REDIS_CONNECTION_STRING');
    expect(previewFileHasPreviewEnv).toBeTruthy();
  });

  it('should handle --environment=production flag', async () => {
    const cwd = setupFixture('vercel-pull-next');
    useUser();
    useTeams('team_dummy');
    useProject({
      ...defaultProject,
      id: 'vercel-pull-next',
      name: 'vercel-pull-next',
    });
    client.setArgv('pull', '--environment=production', cwd);
    const exitCodePromise = pull(client);
    await expect(client.stderr).toOutput(
      'Downloading `production` Environment Variables for Project vercel-pull-next'
    );
    await expect(client.stderr).toOutput(
      `Created .vercel${path.sep}.env.production.local file`
    );
    await expect(client.stderr).toOutput(
      `Downloaded project settings to .vercel${path.sep}project.json`
    );
    await expect(exitCodePromise).resolves.toEqual(0);

    const rawProdEnv = await fs.readFile(
      path.join(cwd, '.vercel', '.env.production.local')
    );
    const previewFileHasPreviewEnv1 = rawProdEnv
      .toString()
      .includes('REDIS_CONNECTION_STRING');
    expect(previewFileHasPreviewEnv1).toBeTruthy();
    const previewFileHasPreviewEnv2 = rawProdEnv
      .toString()
      .includes('SQL_CONNECTION_STRING');
    expect(previewFileHasPreviewEnv2).toBeTruthy();
  });
});
