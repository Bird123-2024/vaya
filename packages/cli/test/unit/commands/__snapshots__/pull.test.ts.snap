// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pull should handle development env vars with comments 1`] = `
"# Created by Vercel CLI
ANOTHER_ENV_STRING="A VALUE GOES HERE"
ANOTHER_ENV_STRING_2="A VALUE GOES HERE_2"
# This connects to postgres on vercel
DATABASE_URL="example.com"
# Connects to the Redis database that stores sessions for fast retrieval. This can be accessed on Vercel KV.
REDIS_CONNECTION_STRING="redis://abc123@redis.example.com:6379"
"
`;

exports[`pull should handle preview env vars with comments 1`] = `
"# Created by Vercel CLI
ANOTHER_ENV_STRING="A VALUE GOES HERE"
ANOTHER_ENV_STRING_2="A VALUE GOES HERE_2"
# This connects to postgres on vercel
DATABASE_URL="example.com"
# Connects to the Redis database that stores sessions for fast retrieval. This can be accessed on Vercel KV.
REDIS_CONNECTION_STRING="redis://abc123@redis.example.com:6379"
"
`;

exports[`pull should handle production env vars with comments 1`] = `
"# Created by Vercel CLI
ANOTHER_ENV_STRING="A VALUE GOES HERE"
ANOTHER_ENV_STRING_2="A VALUE GOES HERE_2"
# This connects to postgres on vercel
DATABASE_URL="example.com"
# Connects to the Redis database that stores sessions for fast retrieval. This can be accessed on Vercel KV.
REDIS_CONNECTION_STRING="redis://abc123@redis.example.com:6379"
"
`;
